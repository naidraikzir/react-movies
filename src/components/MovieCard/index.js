import React from 'react';
import {
  useColorModeValue,
  Box,
  Heading,
  Image
} from '@chakra-ui/react';

const MovieCard = ({
  Poster,
  onPosterClick,
  Title,
  onTitleClick,
}) => {
  const titleBg = useColorModeValue('white', 'gray.800');
  return (
    <Box
      pos="relative"
      cursor="pointer"
    >
      <Image
        src={Poster}
        width="100%"
        height="300px"
        fit="cover"
        rounded="xl"
        onClick={(evt) => {
          evt.stopPropagation();
          onPosterClick();
        }}
      />
      <Heading
        bg={titleBg}
        p="3"
        size="sm"
        pos="absolute"
        bottom="-2em"
        left="1em"
        right="1em"
        rounded="md"
        shadow="lg"
        onClick={onTitleClick}
      >
        {Title}
      </Heading>
    </Box>
  );
};

export default MovieCard;
