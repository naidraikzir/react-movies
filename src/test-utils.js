import React from 'react';
import { render } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { ChakraProvider, theme } from '@chakra-ui/react';
import reducer from 'store/reducer';

const customRender = (
  ui,
  {
    initialState,
    store = createStore(reducer, initialState),
    ...options
  } = {}
) => {
  const AllProviders = ({ children }) => (
    <Provider store={store}>
      <ChakraProvider theme={theme}>{children}</ChakraProvider>
    </Provider>
  );
  return render(ui, { wrapper: AllProviders, ...options });
}

export { customRender as render };
