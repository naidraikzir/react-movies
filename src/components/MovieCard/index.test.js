import React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from 'test-utils';
import MovieCard from '.';

const props = {
  Title: 'Title',
};

test('match snapshot', () => {
  const { container } = render(<MovieCard {...props} />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders title', () => {
  const { getByText } = render(<MovieCard {...props} />);
  expect(getByText(props.Title)).toBeInTheDocument();
});

test('call props onPosterClick on poster click', () => {
  const mockPosterClick = jest.fn();
  const { getByAltText } = render(<MovieCard
    {...props}
    onPosterClick={mockPosterClick}
  />);
  userEvent.click(getByAltText(props.Title));
  expect(mockPosterClick).toHaveBeenCalled();
});

test('call props onTitleClick on title click', () => {
  const mockTitleClick = jest.fn();
  const { getByText } = render(<MovieCard
    {...props}
    onTitleClick={mockTitleClick}
  />);
  userEvent.click(getByText(props.Title));
  expect(mockTitleClick).toHaveBeenCalled();
});
