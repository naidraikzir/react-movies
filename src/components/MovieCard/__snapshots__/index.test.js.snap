// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  class="css-dfpqc0"
>
  <img
    alt="Title"
    class="chakra-image__placeholder css-md0tyy"
  />
  <h2
    class="chakra-heading css-ic2ri5"
  >
    Title
  </h2>
</div>
`;
